sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -367,7 +367,7 @@ case class ListQuery(
     plan.output.head.dataType
   }
   override lazy val resolved: Boolean = childrenResolved && plan.resolved && numCols != -1
-  override def nullable: Boolean = false
+  override def nullable: Boolean = childOutputs.exists(_.nullable)
```

Comment:
One problem is `ListQuery` can't be executed alone so its nullability is useless. We can even throw exception here to make sure no one call `def nullable`.

Shall we fix `In#nullable` instead? If the list is a single-element Seq of `ListQuery`, the nullability should be defined differently.

Reply:
InSubquery.nullable derives from the nullability of its children, which are the values (lhs) and ListQuery (rhs). ListQuery.nullable does affect the results of the unit test I added.

Reply:
Yea this fixes the problem, but it confuses people about what `ListQuery.nullable` means. If we can evaluate `ListQuery` alone, it should return a non-nullable struct containing the query output columns. Since we never evaluate `ListQuery` alone, its nullability is nonsense.

Reply:
Ok, I can try changing it to throw an exception in ListQuery.nullable and change InSubquery.nullable to check ListQuery.childOutputs directly.